src/streaming-services/nrk/NrkApi.ts

```diff
@@ -65,23 +129,29 @@ class _NrkApi extends Api {
 			}
 			let isLastPage = false;
 			let items: Item[] = [];
-			const historyItems: NrkHistoryItem[] = [];
+			const historyItems: NrkProgressItem[] = [];
 			do {
 				const responseText = await Requests.send({
-					url: `${this.HISTORY_API_URL}?pg=${nextPage}`, //TODO figure out if pagination is even supported in the API
+					url: this.HISTORY_API_URL,
 					method: 'GET',
+					headers: {
+						Authorization: 'Bearer ' + this.token,
+					},
 				});
-				const responseJson = JSON.parse(responseText) as NrkHistoryItem[];
-				if (responseJson && responseJson.length > 0) {
-					itemsToLoad -= responseJson.length;
-					historyItems.push(...responseJson);
+				const responseJson = JSON.parse(responseText) as NrkProgressResponse;
+				const { progresses } = responseJson;
+				this.HISTORY_API_URL = this.API_HOST_URL + responseJson._links.next.href;
+				if (progresses.length > 0) {
```

Comment:
I think you mentioned this before, but this causes and endless loop because this condition will always be true (unless the user has 0 items in their history), so it keeps loading the same page over and over again.

To fix this, we should probably keep a list of the progress IDs that have already been fetched, and check for new ones when loading a new page.

Reply:
I think this is no longer the case?